drivers/boston_project_scorecard/app/controllers/boston_project_scorecard/warehouse_reports/scorecards_controller.rb

```diff
@@ -0,0 +1,280 @@
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module BostonProjectScorecard::WarehouseReports
+  class ScorecardsController < ApplicationController
+    include WarehouseReportAuthorization
+    include ArelHelper
+    # TODO: what are the access rules?
+    before_action :set_filter, only: [:index, :create]
+    before_action :set_projects_and_groups, :set_current_reports, only: [:index]
+    before_action :set_report, only: [:show, :edit, :rewind, :complete, :update]
+    before_action :set_history_filter, only: [:history]
+
+    def index
+      start_date = Date.current.prev_month.beginning_of_month
+      end_date = Date.current.prev_month.end_of_month
+      @range = ::Filters::FilterBase.new(start: start_date, end: end_date, user_id: current_user.id)
+    end
+
+    def history
+      scope = filtered_reports_scope.
+        order(id: :desc).
+        preload(project: [:organization, :data_source])
+      @pagy, @reports = pagy(scope, items: 50)
+    end
+
+    private def available_users
+      creator_user_ids = reports_scope.distinct.pluck(:user_id)
+      User.active.where(id: creator_user_ids)
+    end
+    helper_method :available_users
+
+    def show
+      report_id = { report_id: @report.id }
+      @pdf_export = BostonProjectScorecard::DocumentExports::ScorecardExport.new(query_string: report_id.to_query, user: current_user)
+    end
+
+    def create
+      @range = ::Filters::FilterBase.new(user_id: current_user.id)
+      @range.set_from_params(scorecard_filter_params)
+      errors = @range.errors.messages.map { |k, v| "#{k}: #{v.join(', ')}".humanize }
+
+      @project_ids = params[:project]&.keys&.map(&:to_i) || []
+      @project_group_ids = params[:project_group]&.keys&.map(&:to_i) || []
+      @project_ids = project_scope.where(id: @project_ids).pluck(:id) # viewability allowlist
+      @project_group_ids = project_group_scope.where(id: @project_group_ids).pluck(:id) # viewability allowlist
+
+      errors << 'At least one project or group must be selected' if @project_ids.empty? && @project_group_ids.empty?
+
+      if errors.any?
+        flash[:error] = errors.join('<br />'.html_safe)
+        render action: :index
+      else
+        generate_for_projects(@project_ids, @range, current_user) if @project_ids
+        generate_for_groups(@project_group_ids, @range, current_user) if @project_group_ids
+        flash[:notice] = 'Reports queued for processing'
+        redirect_to(boston_project_scorecard_warehouse_reports_scorecards_path(filters: initial_filter_params))
+      end
+    end
+
+    def edit
+    end
+
+    def rewind
+      rewind_workflow
+      redirect_to action: appropriate_action
+    end
+
+    def complete
+      advance_workflow
+      redirect_to action: :show
+    end
+
+    def update
+      @report.update!(scorecard_params)
+      advance_workflow if params[:commit] == workflow_action
+      rewind_workflow if params[:commit] == 'Back'
```

Comment:
Feels like `advance_workflow` and `rewind_workflow` etc, might be better handed off to the report since they just call methods on it.  Maybe `@report.execute_workflow(params[:commit])`?